src/main.rs

```diff
@@ -71,34 +70,31 @@ impl Deck {
         };
     }
 
-    fn shuffle(&self) -> Deck {
-        let mut cards = self.cards.to_vec();
+    fn shuffle(mut self) -> Deck {
         let mut rng = thread_rng();
-        rng.shuffle(&mut cards);
-        return Deck {
-            cards: cards,
-        };
+        rng.shuffle(&mut self.cards);
+        return self;
     }
 }
 
-struct Column {
+struct Column <'a> {
     hidden_index: usize,
-    cards: Vec<Card>,
+    cards: Vec<&'a Card>,
 }
 
-struct Table {
-    columns: Vec<Column>,
+struct Table <'a> {
+    columns: Vec<Column<'a>>,
 }
 
-fn deal(deck: Deck) -> Table {
+fn deal(deck: &Deck) -> Table {
     let mut columns = vec![];
     let mut start = 0;
     let mut end = 1;
     for column_number in 0..7 {
         columns.push(
             Column {
                 hidden_index: column_number,
-                cards: deck.cards[start..end].to_vec(),
+                cards: deck.cards[start..end].iter().collect(),
```

Comment:
`to_vec` attempts to make a copy of the vector and copy its members.
This change turns a `Vec<Card>` into a `Vec<&Card>`. 
Maybe there's a better way.